pkg/store/cache/memcached.go

```diff
@@ -121,10 +121,10 @@ func (c *RemoteIndexCache) FetchMultiPostings(ctx context.Context, blockID ulid.
 // StoreSeries sets the series identified by the ulid and id to the value v.
 // The function enqueues the request and returns immediately: the entry will be
 // asynchronously stored in the cache.
-func (c *RemoteIndexCache) StoreSeries(ctx context.Context, blockID ulid.ULID, id storage.SeriesRef, v []byte) {
+func (c *RemoteIndexCache) StoreSeries(_ context.Context, blockID ulid.ULID, id storage.SeriesRef, v []byte) {
 	key := cacheKey{blockID, cacheKeySeries(id)}.string()
 
-	if err := c.memcached.SetAsync(ctx, key, v, memcachedDefaultTTL); err != nil {
+	if err := c.memcached.SetAsync(context.Background(), key, v, memcachedDefaultTTL); err != nil {
```

Comment:
Potentially here as well, we pass `ctx` to `SetAsync`, put it seems it's either not used or `context.Background()` always. Could be removed from the interface method.